pkg/dao/migration/trigger_modelchange.go

```diff
@@ -37,20 +37,49 @@ CREATE OR REPLACE FUNCTION {{ .Function }}() RETURNS TRIGGER
 AS 
 $$
 DECLARE
-    ids        json;
-    ids_length integer;
-    payload    text;
+    ids             json;
+    ids_length      integer;
+    environment_ids json;
+    project_ids     json;
+    foreign_keys 	jsonb;
+    payload         text;
 BEGIN
     
     -- Select ID list from transition table,
     -- and convert the result to a JSON string array.
     CASE TG_OP
-    WHEN 'INSERT' THEN
-        SELECT array_to_json(array(SELECT id::text FROM new_table)) INTO ids;
-    WHEN 'UPDATE' THEN
+    WHEN 'INSERT', 'UPDATE' THEN
         SELECT array_to_json(array(SELECT id::text FROM new_table)) INTO ids;
+        BEGIN
+            EXECUTE 'SELECT array_to_json(array(SELECT environment_id::text FROM new_table))' INTO environment_ids;
+        EXCEPTION
+            WHEN undefined_column THEN
+                -- Handle the case when the columns don't exist
+                environment_ids := null;
+        END;
+        BEGIN
+            EXECUTE 'SELECT array_to_json(array(SELECT project_id::text FROM new_table))' INTO project_ids;
+        EXCEPTION
+            WHEN undefined_column THEN
+                -- Handle the case when the columns don't exist
+                project_ids := null;
+        END;
     WHEN 'DELETE' THEN
         SELECT array_to_json(array(SELECT id::text FROM old_table)) INTO ids;
+        BEGIN
+            EXECUTE 'SELECT array_to_json(array(SELECT environment_id::text FROM old_table))' INTO environment_ids;
+        EXCEPTION
+            WHEN undefined_column THEN
+                -- Handle the case when the columns don't exist
+                environment_ids := null;
+        END;
+        BEGIN
+            EXECUTE 'SELECT array_to_json(array(SELECT project_id::text FROM old_table))' INTO project_ids;
+        EXCEPTION
+            WHEN undefined_column THEN
+                -- Handle the case when the columns don't exist
+                project_ids := null;
+        END;
     ELSE
```

Comment:
begin...exception... is a way to achieve our goal, but there are three times DSL here, can we fetch the column info from info_schema at first and then select id, project_id, environment_id together?

Reply:
if the above suggestion is available, can we make the payload looks like this?
```
Timestamp   time.Time    `json:"ts"`
Operation   operation    `json:"op"`
TableSchema string       `json:"tb_s"`
TableName   string       `json:"tb_n"`
Rows      []struct{
  ID          object.ID `json:"id"`
  ProjectID   object.ID `json:"project_id"`
  Environment object.ID `json:"environment_Id"`
}  `json:"ids"`
```